doc/source/user/basics.subclassing.rst

```diff
@@ -31,6 +31,37 @@ things like array slicing.  The complications of subclassing ndarray are
 due to the mechanisms numpy has to support these latter two routes of
 instance creation.
 
+When to use subclassing
+=======================
+
+Besides the additional complexities of subclassing a NumPy array, subclasses
+can run into unexpected behaviour because some functions may convert the
+subclass to a baseclass and "forget" any additional information
+associated with the subclass.
+This means that subclassing can be a convenient approach, but should often be
+avoided in favor of using other interoperability approaches described in
+:ref:`Interoperability with NumPy <basics.interoperability>`.
+
+Subclassing can be a good approach if:
+
+* Your use-case is limited:  If your use-case is limited you may not worry
+  about the surpises if a function ignores the subclass.
```

Comment:
Maybe also:

* You are not worried about maintainability or users other than yourself: Sub-classes are faster to write than "duck arrays" that implement everything from scratch. If you don't have external users, you can implement overrides of methods/functions on an "as-needed" basis.

Reply:
I tried to adopt it. I think it is indeed subtly different, but couldn't really find a phrasing that clarifies it, so I decided to just drop the previous bullet.